tests/testthat/test-deterministic.R

```diff
@@ -0,0 +1,313 @@
+context("deterministic")
+
+test_that("Can run the deterministic filter", {
+  dat <- example_sir()
+  p <- particle_nofilter$new(dat$data, dat$model, dat$compare, dat$index)
+  set.seed(1)
+  ll <- p$run(dat$pars$model(dat$pars$initial()))
+  expect_equal(ll, -245.127512965178)
```

Comment:
Is it the same when you run with the same seed? It is exactly to be expected that there are small differences when run with different seeds, there is some exponential noise in the compare to avoid impossible situations (e.g. model value is zero but observed value is non-zero with negative binomial distribution), so the log-likelihood calculation is lightly stochastic.

Reply:
Ed's comment here is correct